python/kvikio/tests/test_aws_s3.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+import multiprocessing as mp
+import os
+import socket
+import time
+from contextlib import contextmanager
+
+import pytest
+
+import kvikio
+import kvikio.defaults
+
+# TODO: remove before PR merge. Trigger CI error if the remote module wasn't built
+import kvikio._lib.remote_handle  # isort: skip
+
+pytestmark = pytest.mark.skipif(
+    not kvikio.is_remote_file_available(),
+    reason="cannot test remote IO, please build KvikIO with with AWS S3 support",
+)
+
+# Notice, we import boto and moto after the `is_remote_file_available` check.
+import boto3  # noqa: E402
+import moto  # noqa: E402
+import moto.server  # noqa: E402
+
+
+@pytest.fixture(scope="session")
+def endpoint_ip():
+    return "127.0.0.1"
+
+
+@pytest.fixture(scope="session")
+def endpoint_port():
+    # Return a free port per worker session.
+    sock = socket.socket()
+    sock.bind(("127.0.0.1", 0))
+    port = sock.getsockname()[1]
+    sock.close()
+    return port
+
+
+@contextmanager
+def ensure_safe_environment_variables():
+    """
+    Get a context manager to safely set environment variables
+    All changes will be undone on close, hence environment variables set
+    within this contextmanager will neither persist nor change global state.
+    """
+    saved_environ = dict(os.environ)
+    try:
+        yield
+    finally:
+        os.environ.clear()
+        os.environ.update(saved_environ)
+
+
+def start_s3_server(ip_address, port):
+    server = moto.server.ThreadedMotoServer(ip_address=ip_address, port=port)
+    server.start()
+    time.sleep(180)
+    print("ThreadedMotoServer shutting down because of timeout (180s)")
+
+
+@pytest.fixture(scope="session")
+def s3_base(endpoint_ip, endpoint_port):
+    """
+    Fixture to set up moto server in separate process
+    """
+    with ensure_safe_environment_variables():
+        # Use fake aws credentials
+        os.environ["AWS_ACCESS_KEY_ID"] = "foobar_key"
+        os.environ["AWS_SECRET_ACCESS_KEY"] = "foobar_secret"
+        os.environ["AWS_SECURITY_TOKEN"] = "foobar_security_token"
+        os.environ["AWS_SESSION_TOKEN"] = "foobar_session_token"
+        os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
```

Comment:
*nit* if you wanted you could use the `monkeypatch` fixture that pytest provides to handle this.

Reply:
Good idea, I just copied the test code from cudf